azurerm/resource_arm_automation_bool_variable_test.go

```diff
@@ -0,0 +1,203 @@
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAutomationBoolVariableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAutomationBoolVariable_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationBoolVariableExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "value", "false"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMAutomationBoolVariable_complete(t *testing.T) {
+	resourceName := "azurerm_automation_bool_variable.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAutomationBoolVariableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAutomationBoolVariable_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationBoolVariableExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "description", "This variable is created by Terraform acceptance test."),
+					resource.TestCheckResourceAttr(resourceName, "value", "true"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMAutomationBoolVariable_basicCompleteUpdate(t *testing.T) {
+	resourceName := "azurerm_automation_bool_variable.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAutomationBoolVariableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAutomationBoolVariable_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationBoolVariableExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "value", "false"),
+				),
+			},
+			{
+				Config: testAccAzureRMAutomationBoolVariable_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationBoolVariableExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "description", "This variable is created by Terraform acceptance test."),
+					resource.TestCheckResourceAttr(resourceName, "value", "true"),
+				),
+			},
+			{
+				Config: testAccAzureRMAutomationBoolVariable_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationBoolVariableExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "value", "false"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMAutomationBoolVariableExists(resourceName string) resource.TestCheckFunc {
```

Comment:
You could probably generalize this test function across the resources?

Reply:
Created common shared function